src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -358,7 +414,7 @@ std::wstring CascadiaSettings::ExpandEnvironmentVariableString(std::wstring_view
     do
     {
         result.resize(requiredSize);
-        requiredSize = ::ExpandEnvironmentStringsW(source.data(), result.data(), result.size());
+        requiredSize = ::ExpandEnvironmentStringsW(source.data(), result.data(), static_cast<DWORD>(result.size()));

```

Comment:
Should this be a `static_cast` or a `gsl::narrow`? One throws if it doesn't fit; perhaps that's behaviour we want?

Reply:
`static_cast<>` is only a compile time check - not runtime.  If we want a runtime error, then `gsl_narrow()`.  Although given the size of a DWORD (32-bit unsigned) that should not be possible given that the [max length of an env var value is ~32K](https://devblogs.microsoft.com/oldnewthing/?p=15083).  Perhaps we should use gsl_narrow_cast() instead?

> gsl::narrow_cast clearly states that conversion can lose data and it is acceptable